### PR TITLE
Add support for automatic prebuilt repo deployments

### PR DESCRIPTION
**Pull request by stevesoltys**
This adds support for automatically deploying the compiled application to a prebuilt repository that can be found here: https://github.com/stevesoltys/backup-prebuilt

It will deploy builds that are run against the `master` and `develop` branches and push any associated tags.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ before_cache:
   - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
   - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
 
-script: ./gradlew check
```
Comment (grote) [inline .travis.yml]: AFAIK, This will remove running the tests which is probably not intentional.

Comment (stevesoltys) [reply .travis.yml]: I think the tests still run (can look at the build for this PR). I left the `check` task here, and added `assemble`.

Comment (grote) [reply .travis.yml]: Oh sorry, I had missed that since Github highlighted the `assemble` part.

File: deploy-prebuilt.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+git checkout $TRAVIS_BRANCH || git checkout -b $TRAVIS_BRANCH
+rm -Rf ./*
+cp $TRAVIS_BUILD_DIR/Android.mk .
+cp $TRAVIS_BUILD_DIR/app/build/outputs/apk/release/app-release-unsigned.apk ./Backup.apk
+cp $TRAVIS_BUILD_DIR/permissions_com.stevesoltys.backup.xml .
+cp $TRAVIS_BUILD_DIR/whitelist_com.stevesoltys.backup.xml .
+cp $TRAVIS_BUILD_DIR/default-permissions_com.stevesoltys.backup.xml .
+
+git add .
+git commit -m "Travis build $TRAVIS_BUILD_NUMBER"
```
Comment (grote) [inline deploy-prebuilt.sh]: @stevesoltys Maybe it would be nice if we'd include a link to the commit that was used for the build?

Comment (stevesoltys) [reply deploy-prebuilt.sh]: Opened a PR for that: https://github.com/stevesoltys/backup/pull/43

**Pull request merged**